driver-lambda/src/main/com/mongodb/lambdatest/LambdaTestApp.java

```diff
@@ -0,0 +1,143 @@
+                .addCommandListener(new CommandListener() {
+                    @Override
+                    public void commandSucceeded(@NonNull final CommandSucceededEvent event) {
+                        totalCommandCount++;
+                        totalCommandDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                    @Override
+                    public void commandFailed(@NonNull final CommandFailedEvent event) {
+                        totalCommandCount++;
+                        totalCommandDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                })
+                .applyToServerSettings(builder -> builder.addServerMonitorListener(new ServerMonitorListener() {
+                    @Override
+                    public void serverHeartbeatSucceeded(@NonNull final ServerHeartbeatSucceededEvent event) {
+                        totalHeartbeatCount++;
+                        totalHeartbeatDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                    @Override
+                    public void serverHeartbeatFailed(@NonNull final ServerHeartbeatFailedEvent event) {
+                        totalHeartbeatCount++;
+                        totalHeartbeatDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                }))
+                .applyToConnectionPoolSettings(builder -> builder.addConnectionPoolListener(new ConnectionPoolListener() {
+                    @Override
+                    public void connectionCreated(@NonNull final ConnectionCreatedEvent event) {
+                        openConnections++;
+                    }
+                    @Override
+                    public void connectionClosed(@NonNull final ConnectionClosedEvent event) {
+                        openConnections--;
+                    }
+                }))
+                .build();
+        mongoClient = MongoClients.create(settings);
+    }
+
+    public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
+        try {
+            MongoCollection<Document> collection = mongoClient
+                    .getDatabase("lambdaTest")
+                    .getCollection("test");
+            BsonValue id = collection.insertOne(new Document("n", 1)).getInsertedId();
+            collection.deleteOne(new Document("_id", id));
+
+            BsonDocument responseBody = new BsonDocument()
+                    .append("totalCommandDuration", new BsonDouble(totalCommandDuration))
+                    .append("totalCommandCount", new BsonDouble(totalCommandCount))
+                    .append("totalHeartbeatDuration", new BsonDouble(totalHeartbeatDuration))
+                    .append("totalHeartbeatCount", new BsonDouble(totalHeartbeatCount))
+                    .append("openConnections", new BsonDouble(openConnections));
```

Comment:
Note possible differences in naming vs [other drivers](https://github.com/mongodb/node-mongodb-native/commit/ce8e69b6db0eb4c0b9271ce43e334d70b967d227#diff-d63b102d9c9e785a60bfe991630209e6f855b532610cf4aae1f9c2fe71b98164R66). Happy to change to match (though perhaps in that case the [spec](https://github.com/mongodb/specifications/blob/master/source/faas-automated-testing/faas-automated-testing.rst#implementing-the-function) could specify these exactly.)

Reply:
Don't think the exact naming is important as we aren't asserting anything.